vmdb/app/models/mixins/ems_openstack_mixin.rb

```diff
@@ -24,7 +24,7 @@ def browser_url
   def openstack_handle(options = {})
     require 'openstack/openstack_handle'
     @openstack_handle ||= begin
-      raise "no credentials defined" if self.missing_credentials?(options[:auth_type])
+      raise MiqException::MiqInvalidCredentialsError, "No credentials defined" if self.missing_credentials?(options[:auth_type])
```

Comment:
But then again, I think MiqInvalidCredentialsError is good for this case...it's up to you @jrafanie